File: frontend/src/utils/timeUtils.js
/**
 * Format duration in minutes to human-readable format
 * @param {number} minutes - Duration in minutes
 * @returns {string} Formatted duration string (e.g., "2h 7m", "1d 3h", "45m")
 */
export const formatDuration = (minutes) => {
  if (!minutes || minutes < 0) {
    return "0m";
  }
  
  if (minutes < 60) {
    return `${Math.floor(minutes)}m`;
  } else if (minutes < 1440) { // Less than 24 hours
    const hours = Math.floor(minutes / 60);
    const remainingMinutes = Math.floor(minutes % 60);
    if (remainingMinutes === 0) {
      return `${hours}h`;
    } else {
      return `${hours}h ${remainingMinutes}m`;
    }
  } else { // 24+ hours
    const days = Math.floor(minutes / 1440);
    const remainingHours = Math.floor((minutes % 1440) / 60);
    const remainingMinutes = Math.floor(minutes % 60);
    
    const parts = [`${days}d`];
    if (remainingHours > 0) {
      parts.push(`${remainingHours}h`);
    }
    if (remainingMinutes > 0) {
      parts.push(`${remainingMinutes}m`);
    }
    
    return parts.join(" ");
  }
};

/**
 * Format duration with both human-readable and exact values
 * @param {number} minutes - Duration in minutes
 * @returns {object} Object with 'formatted', 'minutes', and 'exact' keys
 */
export const formatDurationDetailed = (minutes) => {
  return {
    formatted: formatDuration(minutes),
    minutes: minutes,
    exact: `${minutes} minutes`
  };
};

File: src/utils/time_utils.py
"""
Time utility functions for formatting durations and timestamps
"""

def format_duration(minutes: int) -> str:
    """
    Format duration in minutes to human-readable format
    
    Args:
        minutes: Duration in minutes
        
    Returns:
        Formatted duration string (e.g., "2h 7m", "1d 3h", "45m")
    """
    if minutes < 0:
        return "0m"
    
    if minutes < 60:
        return f"{minutes}m"
    elif minutes < 1440:  # Less than 24 hours
        hours = minutes // 60
        remaining_minutes = minutes % 60
        if remaining_minutes == 0:
            return f"{hours}h"
        else:
            return f"{hours}h {remaining_minutes}m"
    else:  # 24+ hours
        days = minutes // 1440
        remaining_hours = (minutes % 1440) // 60
        remaining_minutes = minutes % 60
        
        parts = [f"{days}d"]
        if remaining_hours > 0:
            parts.append(f"{remaining_hours}h")
        if remaining_minutes > 0:
            parts.append(f"{remaining_minutes}m")
        
        return " ".join(parts)

def format_duration_detailed(minutes: int) -> dict:
    """
    Format duration with both human-readable and exact values
    
    Args:
        minutes: Duration in minutes
        
    Returns:
        Dict with 'formatted', 'minutes', and 'exact' keys
    """
    return {
        'formatted': format_duration(minutes),
        'minutes': minutes,
        'exact': f"{minutes} minutes"
    }
